sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -3402,6 +3402,53 @@ object GpuOverrides extends Logging {
           GpuVarianceSamp(childExprs.head, !legacyStatisticalAggregate)
         }
       }),
+    expr[Percentile](
+      "Aggregation computing exact percentile",
+      ExprChecks.reductionAndGroupByAgg(
+        // The output can be a single number or array depending on whether percentiles param
+        // is a single number or an array.
+        TypeSig.gpuNumeric + TypeSig.ARRAY.nested(TypeSig.gpuNumeric),
+        TypeSig.cpuNumeric + TypeSig.ARRAY.nested(TypeSig.cpuNumeric),
+        Seq(
+          ParamCheck("input",
+            TypeSig.gpuNumeric,
+            TypeSig.cpuNumeric),
+          ParamCheck("percentage",
+            TypeSig.DOUBLE + TypeSig.ARRAY.nested(TypeSig.DOUBLE),
+            TypeSig.DOUBLE + TypeSig.ARRAY.nested(TypeSig.DOUBLE)),
+          ParamCheck("frequency",
+            TypeSig.LONG + TypeSig.ARRAY.nested(TypeSig.LONG),
+            TypeSig.LONG + TypeSig.ARRAY.nested(TypeSig.LONG)))),
+      (c, conf, p, r) => new TypedImperativeAggExprMeta[Percentile](c, conf, p, r) {
+        override def convertToGpu(childExprs: Seq[Expression]): GpuExpression = {
+          val exprMeta = p.get.asInstanceOf[BaseExprMeta[_]]
+          val context = exprMeta.context
+          context match {
+            case ReductionAggExprContext => GpuPercentile(childExprs.head,
+              childExprs(1).asInstanceOf[GpuLiteral],
+              childExprs(2),
+              isReduction = true)
+            case GroupByAggExprContext => GpuPercentile(childExprs.head,
+              childExprs(1).asInstanceOf[GpuLiteral],
+              childExprs(2),
+              isReduction = false)
+            case _ => throw new IllegalStateException(s"Invalid aggregation context: $context")
+          }
+        }
+        // Declare the data type of the internal buffer so it can be serialized and
+        // deserialized correctly during shuffling.
+        override def aggBufferAttribute: AttributeReference = {
+          val aggBuffer = c.aggBufferAttributes.head
+          val dataType: DataType = ArrayType(StructType(Seq(
```

Comment:
We should have a follow on to support the conversion if needed, because both are using maps and it should not be that hard.